lib/synapse/service_watcher/base.rb

```diff
@@ -102,15 +120,16 @@ def validate_discovery_opts
       log.warn "synapse: warning: a stub watcher with no default servers is pretty useless" if @default_servers.empty?
     end
 
-    def filter_backends_by_label(backends, label_filter)
-      filtered_backends = []
-      backends.each do |backend|
+    def backends_filtered_by_labels
+      filtered_backends = @backends.select do |backend|
         backend_labels = backend['labels'] || {}
-        if label_filter['condition'] == 'equals' and backend_labels[label_filter['label']] == label_filter['value']
-          filtered_backends << backend
+        @label_filters.all? do |label_filter|
+          (label_filter['condition'] == 'equals' &&
+            backend_labels[label_filter['label']] == label_filter['value']) ||
+          (label_filter['condition'] == 'not-equals' &&
```

Comment:
hmmm... good point. i wonder if i should explicitly check whether `backend_labels.include?(label_filter['label'])` before matching anything?